numpy/core/src/umath/ufunc_object.c

```diff
@@ -924,30 +935,67 @@ convert_ufunc_arguments(PyUFuncObject *ufunc,
     PyObject *obj;
 
     /* Convert and fill in input arguments */
+    int all_scalar = 1;
```

Comment:
I've wondered more generally, but when I was contributing, I was at some point told to use `npy_bool` for cases like this. I think this gives a bit of clarity, though really it doesn't matter much!

Reply:
Hmm, yeah... We even can use C99 bools.  Mainly a missing habit for me, will change it here (but not everywhere yet).